apps/framework-cli/src/cli/routines.rs

```diff
@@ -628,3 +638,153 @@ pub async fn remote_plan(
+        InfrastructureMap::load_from_user_code(project).await?
+    } else {
+        debug!("Loading InfrastructureMap from primitives");
+        let primitive_map = PrimitiveMap::load(project).await?;
+        InfrastructureMap::new(project, primitive_map)
+    };
+
+    // Get authentication token - prioritize command line parameter, then env var, then project config
+    let auth_token = token
+        .clone()
+        .or_else(|| std::env::var("MOOSE_ADMIN_TOKEN").ok())
+        .ok_or_else(|| anyhow::anyhow!("Authentication token required. Please provide token via --token parameter or MOOSE_ADMIN_TOKEN environment variable"))?;
+
+    let client = reqwest::Client::new();
+
+    let reality_check_url = prepend_base_url(base_url.as_deref(), "admin/reality-check");
+    display::show_message_wrapper(
+        MessageType::Info,
+        Message {
+            action: "Remote State".to_string(),
+            details: format!("Checking database state at {}", reality_check_url),
+        },
+    );
+
+    let response = client
+        .get(&reality_check_url)
+        .header("Authorization", format!("Bearer {}", auth_token))
+        .send()
+        .await?;
+
+    if !response.status().is_success() {
+        let error_text = response.text().await?;
+        return Err(anyhow::anyhow!(
+            "Failed to get reality check from remote instance: {}",
+            error_text
+        ));
+    }
+
+    #[derive(Deserialize)]
+    struct RealityCheckResponse {
+        discrepancies: InfraDiscrepancies,
+    }
+
+    let reality_check: RealityCheckResponse = response.json().await?;
+    debug!("Remote discrepancies: {:?}", reality_check.discrepancies);
+
+    // Step 3: Find tables that exist both in local infra map and remote tables
+    let mut tables_to_integrate = Vec::new();
+
+    fn warn_about_mismatch(table_name: &str) {
+        display::show_message_wrapper(
+            MessageType::Highlight,
+            Message {
+                action: "Table".to_string(),
+                details: format!(
+                    "Table {} in remote DB differs from local definition. It will not be integrated.",
+                    table_name,
+                ),
+            },
+        );
+    }
+
+    // Check unmapped tables
+    for table in &reality_check.discrepancies.unmapped_tables {
+        if let Some(local_table) = local_infra_map
+            .tables
+            .values()
+            .find(|t| t.name == table.name)
+        {
+            match InfrastructureMap::diff_table(table, local_table) {
+                None => {
+                    debug!("Found matching table: {}", table.name);
+                    tables_to_integrate.push(table.name.clone());
+                }
+                Some(_) => warn_about_mismatch(&table.name),
+            }
+        }
+    }
+
+    // Check mismatched tables
+    for change in &reality_check.discrepancies.mismatched_tables {
```

Comment:
Can you explain why you run this on the mismatched tables?

I was expecting it to skip given the `warn_about_mismatch` function above

Reply:
warn about "local" and "remote reality" mismatch

"remote infra map" and "remote reality" mismatch is fine

Reply:
ie you integrate the changes if they are between remote reality and remote infra map. But you don't if they are between local and remote reality

if that's it would be amazing to make that explicit in the comments